lucene/core/src/java/org/apache/lucene/search/TotalHitCountCollector.java

```diff
@@ -16,7 +16,12 @@
  */
 package org.apache.lucene.search;
 
-/** Just counts the total number of hits. */
+/**
+ * Just counts the total number of hits. For cases when this is the only collector used, {@link
+ * IndexSearcher#count(Query)} should be called instead of {@link IndexSearcher#search(Query,
+ * Collector)} as the former is faster whenever the match can be returned directly from the index
```

Comment:
Did you mean s/match/count/?

```suggestion
 * Collector)} as the former is faster whenever the count can be returned directly from the index
```

Reply:
I meant match count, but count is fine too.